app/resources/v1/users.js

```diff
@@ -110,6 +111,28 @@ exports.post = function (req, res) {
     return nickname + '-' + id
   }
 
+  const handleUserProfileImage = async function (user, credentials) {
+    const publicId = `${config.env}/profile_image/${user.id}`
+    let profileImageUrl
+
+    // Check if user has profile image already cached in cloudinary
+    if (user.profile_image_url && user.profile_image_url.includes(publicId)) {
+      profileImageUrl = user.profile_image_url
+    } else if (credentials.profile_image_url) {
+      // If no profile image cached in cloudinary, cache image provided by credentials and return cloudinary url.
+      try {
+        const response = await cloudinary.v2.uploader.upload(credentials.profile_image_url, { upload_preset: 'profile_image', public_id: publicId })
+        profileImageUrl = response.secure_url
+      } catch (error) {
```

Comment:
Similarly to the comment in `app/resources/v1/user.js`, I'm curious what the value of `error` is when an error happens. There's usually two types of errors that can happen in this context, an HTTP error (like a 404 or a 500) or a syntax error (like `response` is undefined). Let's say there's a 500 error on Cloudinary's side. The Cloudinary client library can do one of three things:

1. Doesn't throw an error, just passes the result as response. The response might look something like `{ "error": true, "statusCode": 500 }`. Still, the `try` block will succeed (because nothing technically broke) and `profileImageUrl` will be set to `undefined` because `secure_url` isn't present.
2. Throws an error, and hopefully the `error` variable contains enough useful information that we can track down what went wrong later. 
3. Doesn't throw an error, but the result is of an unexpected shape, so code breaks elsewhere outside of the Cloudinary client library. Say for instance that `response` ends up returned as `undefined`. In this case `error` will be something like "Unable to read `secure_url` of undefined". It gets logged and we get no information about why `response` is undefined.

Hopefully that makes sense, and hopefully you can point me at some Cloudinary doc that makes it clear! (Or we can work on tests that make it clear what the expected return values are for errors!)